mdvt/contribute/util.py

```diff
@@ -89,84 +93,145 @@ def api_info_url(pageid):
-            file_depicts = get_file_depicts(page['title'])
-            if file_depicts is not None:
-                (depict_id, depict_label,
-                 depict_description, claim_id) = file_depicts
-            else:
-                continue
 
-            contribute_request = {
-                'media_page': page['fullurl'],
-                'media_page_id': page['pageid'],
-                'media_title': page['title'],
-                'depict_id': depict_id,
-                'depict_label': depict_label,
-                'depict_description': depict_description,
-                'claim_id': claim_id,
-                'csrf': gen_csrf()
-            }
-            return contribute_request
-    elif filter_type == 'tag':
-        changes, continue_key = api_tagged_changes(filter_value, continue_key)
+def gen_csrf(question_id):
+    csrf = secrets.token_hex(16)
+    session['csrf'] = [question_id, csrf]
+    return csrf
 
-        for change in changes:
-            if change['ns'] != 6:
-                continue
-            file_depicts = get_file_depicts(change['title'])
-            if file_depicts is not None:
-                (depict_id, depict_label,
-                 depict_description, claim_id) = file_depicts
-            else:
-                continue
 
-            contribute_request = {
-                'media_page': api_info_url(str(change['pageid'])),
-                'media_page_id': change['pageid'],
-                'media_title': change['title'],
-                'depict_id': depict_id,
-                'depict_label': depict_label,
-                'depict_description': depict_description,
-                'claim_id': claim_id,
-                'csrf': gen_csrf()
+# TODO: solve 414 or use id instead of titles
+def get_questions(filter_type, filter_value, continue_key=None):
+    true_count = 0
+    false_count = 0
+    skip_count = 0
+    questions = (Question.query
+                 .filter(Question.filter_type == filter_type)
+                 .filter(Question.filter_value == filter_value)
+                 .all())
+    for question in questions:
+        true_count = (Contribution.query
+                      .filter(Contribution.question_id == question.id)
+                      .filter(Contribution.answer == 'true')
+                      .count())
+        false_count = (Contribution.query
+                       .filter(Contribution.question_id == question.id)
+                       .filter(Contribution.answer == 'false')
+                       .count())
+        skip_count = (Contribution.query
+                      .filter(Contribution.question_id == question.id)
+                      .filter(Contribution.answer == 'skip')
+                      .filter(Contribution.user_id == session['user_id'])
+                      .count())
+        if not true_count and not false_count and not skip_count:
+            page = requests.get(
+                'https://commons.wikimedia.org/w/api.php',
```

Comment:
```suggestion
                'https://commons.wikimedia.org/w/api.php',
```
Can we define this in a config file somewhere and use it?

Reply:
solved in commit ed2cba6 :)

Reply:
Great! My advise will be keep the commits atomic and an detailed.